### PR TITLE
Encounter updates

### PR DESCRIPTION
**Pull request by sjpadgett**
- set POS for appointment auto encounter

Working #3449

below added by bradymiller:
Up for Grabs demo for this PR is here: https://www.open-emr.org/wiki/index.php/Development_Demo#Alpha_-_Up_For_Grabs_Demo

### CODE REVIEW

File: interface/patient_file/encounter/forms.php [hunk 1]
```diff
@@ -91,6 +91,12 @@
 <script type="text/javascript">
 $.noConflict();
 jQuery(document).ready( function($) {
+    <?php if ((boolean) $GLOBALS['expand_form']) { ?>
```
Comment (bradymiller) [inline interface/patient_file/encounter/forms.php:94]: @sjpadgett , Thanks for fixing this. We should prob make expand_form default to on in globals.inc.php.

Comment (bradymiller) [reply interface/patient_file/encounter/forms.php:94]: oh wait, disregard that, I didn't realize this was rel-502

Comment (sjpadgett) [reply interface/patient_file/encounter/forms.php:94]: Looks like it is fixed in 5.0.3 but i've not looked at what it defaults to.

**Comment by sjpadgett** [comment]
Getting close. @stephenwaite You can beat it up if you want or @bradymiller can put up a up for grab.
Encounters seem to being doing a better job at tracking with events.

File: library/ajax/facility_ajax_code.php [hunk 2]
```diff
@@ -22,16 +22,26 @@
...
-    echo ((int)$pos['pos_code'] < 10) ? ("0" . $pos['pos_code']) : $pos['pos_code'];
+
+    echo json_encode(((int)$pos['pos_code'] < 10) ? ("0" . $pos['pos_code']) : $pos['pos_code']);
     exit();
-}
+} elseif ($_GET['mode'] === 'get_user_data') {
+    // put here for encounter user changes sjp
+    $provider_id = $_GET['provider_id'] ? (int)$_GET['provider_id'] : exit('0');
+    $userService = new \OpenEMR\Services\UserService();
+    $facilityService = new \OpenEMR\Services\FacilityService();
```
Comment (bradymiller) [inline library/ajax/facility_ajax_code.php:36]: rec migrating the namespace paths up to the `use` block at top of script

**Comment by bradymiller** [comment]
Up for Grabs demo for this PR is here: https://www.open-emr.org/wiki/index.php/Development_Demo#Alpha_-_Up_For_Grabs_Demo

**Comment by bradymiller** [comment]
oops, mistakenly closed/reopened and triggered our foe, travis  :fearful:

**Comment by sjpadgett** [comment]
> oops, mistakenly closed/reopened and triggered our foe, travis

umm since when is he OUR foe. Your the one always contributing to his delinquency!:)

For v5.0.2 I don't see class paths changing but, i'll change. I'm still trying to understand why I take on this stuff. It's not like I don't have ten other things to do!
Maybe trying to repent for something. Who knows:)

**Comment by bradymiller** [comment]
There always needs to be a foe, and travis is the perfectly engineered scapegoat for this. So, instead of thinking, "I'm an idiot", I just think "travis is evil" instead. Helps to avoid needing therapy :)
Regarding taking on painful projects, I am also guilty of that.

**Comment by sjpadgett** [comment]
Well this has been bugging me anyways. I just wish I had some practical clinical knowledge to help guide me in reasoning some of this stuff out. Still, we'll do what we can.....

> I just think "travis is evil"

My thoughts exactly! The thought of a group of who knows who deciding what is appropriate stylings drives me insane with many times the requirements are just stupid. So, all now know how I really feel! However, you already know:)

**Comment by sjpadgett** [comment]
I still want to try to get one more fix with this PR then we prob should get on a security patch....
I was thinking we might want to think about adding a security patch level for tracking. Also we may want to have feature update patches as well.

**Comment by sjpadgett** [comment]
Bringing in so I can forward port to 5.0.3! Leave it to me to do this backwards.:)

**Pull request merged**